tools/deploy.go

```diff
@@ -0,0 +1,591 @@
+	lines, err := readLines(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read lines from file %s: %v", filePath, err)
+	}
+
+	updatedLines := []string{}
+	found := false
+	for _, line := range lines {
+		if strings.HasPrefix(line, key+"=") {
+			updatedLines = append(updatedLines, fmt.Sprintf("%s=%s", key, value))
+			found = true
+		} else {
+			updatedLines = append(updatedLines, line)
+		}
+	}
+	if !found {
+		updatedLines = append(updatedLines, fmt.Sprintf("%s=%s", key, value))
+	}
+
+	err = writeLines(updatedLines, filePath)
+	if err != nil {
+		return fmt.Errorf("failed to write updated lines to file %s: %v", filePath, err)
+	}
+
+	return nil
+}
+
+func validateCmd(cmd []string) error {
+	for _, arg := range cmd {
+		if strings.Contains(arg, ";") || strings.Contains(arg, "&&") || strings.Contains(arg, "|") {
+			return errors.New("invalid character in command arguments")
+		}
+	}
+
+	return nil
+}
+
+// pipeCommands pipes the output of cmd1 to the input of cmd2.
+func pipeCommands(cmd1 *exec.Cmd, baseCmd []string) error {
+	err := validateCmd(baseCmd)
+	if err != nil {
+		return err
+	}
+
+	cmd2 := exec.Command(baseCmd[0], baseCmd[1:]...) // #nosec G204:gosec
+
+	pipeOut, err := cmd1.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	defer pipeOut.Close()
+
+	cmd2.Stdin = pipeOut
+
+	cmd1.Stderr = os.Stderr
+	cmd2.Stdout = os.Stdout
+	cmd2.Stderr = os.Stderr
+
+	err = cmd1.Start()
+	if err != nil {
+		return err
+	}
+
+	err = cmd2.Start()
+	if err != nil {
+		return err
+	}
+
+	err = cmd1.Wait()
+	if err != nil {
+		return err
+	}
+
+	err = cmd2.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func copyFile(src, dst string) error {
+	input, err := os.ReadFile(src)
+	if err != nil {
+		return fmt.Errorf("failed to read source file %s: %v", src, err)
+	}
+
+	err = os.WriteFile(dst, input, 0600)
+	if err != nil {
+		return fmt.Errorf("failed to write destination file %s: %v", dst, err)
+	}
+
+	return nil
+}
+
+func prepareKubectlCmdArgs(baseCmd []string, kubectlArgs string) []string {
+	if kubectlArgs != "" {
+		baseCmd = append(baseCmd, strings.Split(kubectlArgs, " ")...)
+	}
+
+	return append(baseCmd, "-f", "-")
+}
+
+func deployBMO(tempBMOOverlay, scriptDir, kubectlArgs, kustomizePath string) error {
+	revertFunc, err := changeDirTemporarily(tempBMOOverlay)
+	if err != nil {
+		log.Fatalf("Failed to change directory %s: %v", tempBMOOverlay, err)
+	}
+	defer revertFunc()
+
+	// This is to keep the current behavior of using the ironic.env file for the configmap
+	ironicEnvSrc := filepath.Join(scriptDir, "config", "default", "ironic.env")
+	ironicEnvDst := filepath.Join(tempBMOOverlay, "ironic.env")
+	copyFile(ironicEnvSrc, ironicEnvDst)
+
+	execCommand(kustomizePath, "edit", "add", "configmap", "ironic", "--behavior=create", "--from-env-file=ironic.env")
+
+	kubectlBaseCmd := []string{"kubectl", "apply"}
+	kubectlBaseCmd = prepareKubectlCmdArgs(kubectlBaseCmd, kubectlArgs)
```

Comment:
Can we use k8s client go libraries for kubectl related commands?

Reply:
Thanks for the feedback, I'll look into it!